docs/shared/features/enhance-tooling.md

```diff
@@ -0,0 +1,102 @@
+# Enhance Tooling
```

Comment:
I'm missing more info on what gets enhanced and the benefit of it, like why would I want it. 

When I read this page I get some quick "you get caching, task pipelines,.." which as mentioned in the next comment is not exclusive to the plugins. I get how to set it up via `nx init`, how to use it and types of targets as well as how to modify options (where I'm not even sure I'd need the two here 🤔..but let's leave it for now).
But not really what that enhancing gives me